src/components/Card.js

```diff
@@ -1,38 +1,63 @@
 class Card {
-  constructor({ name, link }, cardSelector, handleImageClick) {
+  constructor(
+    { name, link, cardID, isLiked, _id, cardData },
+    cardSelector,
+    handleImageClick,
+    handleLikeClick,
+    handleDeleteSubmit
+  ) {
     this._name = name;
     this._link = link;
+    this._cardID = cardID;
+    this._id = _id;
+    this.isLiked = isLiked;
+    this.cardData = cardData;
     this._cardSelector = cardSelector;
     this._handleImageClick = handleImageClick;
+    this._handleLikeClick = handleLikeClick;
+    this._handleDeleteSubmit = handleDeleteSubmit;
   }
 
   _setEventListeners() {
-    this._cardElement
-      .querySelector(".card__like-button")
-      .addEventListener("click", () => {
-        this._handleLikeButton();
-      });
-
-    this._cardElement
-      .querySelector(".card__delete-button")
-      .addEventListener("click", () => {
-        this._handleDeleteButton();
-      });
+    this._likeButton = this._cardElement.querySelector(".card__like-button");
+
+    this._likeButton.addEventListener("click", () => {
+      this._handleLikeClick(this);
+    });
+
+    this._deleteButton.addEventListener("click", () => {
+      this._handleDeleteSubmit(this._id, this);
+    });
 
     this._cardImage.addEventListener("click", () => {
       this._handleImageClick({ link: this._link, name: this._name });
     });
   }
 
-  _handleDeleteButton() {
+  handleDeleteSubmit() {
+    const cardID = this.cardData;
+    handleDeleteButton(cardID);
+  }
+
+  isLiked() {
+    return this.isLiked;
+  }
+
+  _setLikes() {
```

Comment:
Needs correction: please rename this method to something like `_renderLikes`, since it's responsible for likes rendering